daemons/pacemakerd/pcmkd_messages.c

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2010-2021 the Pacemaker project contributors
+ *
+ * The version control history for this file may have further details.
+ *
+ * This source code is licensed under the GNU General Public License version 2
+ * or later (GPLv2+) WITHOUT ANY WARRANTY.
+ */
+
+#include <crm_internal.h>
+#include "pacemakerd.h"
+
+crm_exit_t
+request_shutdown(crm_ipc_t *ipc)
```

Comment:
That might be a candidate for a library-function.

Reply:
Good point, that would make a nice addition to bring pacemakerd fully in line with the new IPC model and expose @clumens to it.

pacemakerd's old_instance would need to follow the new model, using pcmk_new_ipc_api() instead of crm_ipc_new() (crm_mon.c has an existing use for comparison, though pacemakerd would probably use pcmk_ipc_dispatch_sync). The new function would go in lib/common/ipc_pacemakerd.c (comparable to pcmk_pacemakerd_api_ping()), and reply_expected() there would need to be updated to include it.

That can be a separate PR